node/channel.js

```diff
@@ -721,17 +720,18 @@ TChannel.prototype._request = function _request(opts) {
         req = opts.peer.request(opts);
     } else if (opts.host) {
         opts.retryCount = 0;
-        req = self.peers.add(opts.host).request(opts);
+        opts.peer = self.peers.add(opts.host);
```

Comment:
Not sure why the peer needs to be assigned to opts. It doesn't seem a good idea to change the request parameter.